### PR TITLE
Added special catch for std::exception in GTEST_TEST_NO_THROW_

### PR DESCRIPTION
**Pull request by Jonny007-MKD**
Just like #911 this PR aims to output the message of a std::exception in case of a failure.

### CODE REVIEW

**Comment by Jonny007-MKD** [comment]
I signed it!

**Comment by Jonny007-MKD** [comment]
I signed it!

**Comment by Jonny007-MKD** [comment]
I signed it!

File: googletest/test/gtest_unittest.cc [hunk 1]
```diff
@@ -3812,6 +3815,9 @@ TEST(AssertionTest, ASSERT_NO_THROW) {
   EXPECT_FATAL_FAILURE(ASSERT_NO_THROW(ThrowAnInteger()),
                        "Expected: ThrowAnInteger() doesn't throw an exception."
                        "\n  Actual: it throws.");
+  EXPECT_FATAL_FAILURE(ASSERT_NO_THROW(ThrowAnException("blablubb")),
+                       "Expected: ThrowAnException(\"blablubb\") doesn't throw an exception."
```
Comment (JonathanDCohen) [inline googletest/test/gtest_unittest.cc]: This looks overspecified.  The important bit is that we have what was actually thrown in the message.

File: googletest/include/gtest/internal/gtest-internal.h [hunk 2]
```diff
@@ -1242,22 +1242,22 @@ class AdditionalMessage
...
-            "with message: " + std::string(e.what()); \
+            "with message: " + std::string(e.what())); \
         goto GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__); \
       } \
     } \
     catch (...) { \
       if (!gtest_caught_expected) { \
-        message = \
-            "it throws a different type."; \
+        message.set( \
```
[the remark this answers is not in the document]
Comment (Jonny007-MKD) [reply googletest/include/gtest/internal/gtest-internal.h]: ok, done

**Comment by gennadiycivil** [comment]
Revering, due to style violations

**Comment by gennadiycivil** [comment]
I was checking unrelated code and noticed that even though this PR has been merged there are sill code style violations flagged by the various tools internally. ( In particular 80 columns wide limits and the name "value" . 
I am attempting to revert this PR

**Comment by Jonny007-MKD** [comment]
sure, no problem :) seems like I have to rebranch and recommit the changes, right?

**Comment by gennadiycivil** [comment]
Yup, just do the same PR again ( sorry )

**Pull request merged**